### PR TITLE
modifications to business rule

### PR DESCRIPTION
**Pull request by AnuGayan**

### CODE REVIEW

File: components/org.wso2.carbon.business.rules.core/src/test/java/org/wso2/carbon/business/rules/core/internal/TemplateManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package org.wso2.carbon.business.rules.core.internal;
```
Comment (Anoukh) [inline components/org.wso2.carbon.business.rules.core/src/test/java/org/wso2/carbon/business/rules/core/internal/TemplateManagerTest.java]: Missing licence header?

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/services/TemplateManagerService.java [hunk 2]
```diff
@@ -107,7 +105,7 @@ public void deleteBusinessRule(String uuid) { // todo: verify next lower level
             foundBusinessRule = findBusinessRuleFromTemplate(uuid);
         } catch (TemplateManagerException e) {
             // No Business Rule Found
-            log.error(e); // todo: (Q) is this ok?
+            log.error(e.toString()); // todo: (Q) is this ok?
```
Comment (mohanvive) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/services/TemplateManagerService.java]: Please provide more context info for error. Check other places as well.

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/services/TemplateManagerService.java [hunk 3]
```diff
@@ -172,7 +170,9 @@ public void deployTemplates(BusinessRuleFromTemplate businessRuleFromTemplate) {
                 // If file is a valid json file
                 if (fileEntry.isFile() && fileEntry.getName().endsWith("json")) {
                     // convert and store
-                    TemplateGroup templateGroup = TemplateManagerHelper.jsonToTemplateGroup(TemplateManagerHelper.fileToJson(fileEntry));
+                    TemplateGroup templateGroup = TemplateManagerHelper.jsonToTemplateGroup(TemplateManagerHelper
+                            .fileToJson
+                            (fileEntry));
```
Comment (mohanvive) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/services/TemplateManagerService.java]: Format code properlly.

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/bean/businessRulesFromTemplate/BusinessRuleFromTemplate.java [hunk 4]
```diff
@@ -27,7 +27,7 @@
  */
 public class BusinessRuleFromTemplate extends BusinessRule {
     private String ruleTemplateName;
-    private Map<String, String> properties;
+    private Map<String, String> properties; // TODO: 9/17/17  BusinessRulePropertyFromTemplate
```
Comment (dnwick) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/bean/businessRulesFromTemplate/BusinessRuleFromTemplate.java]: remove todo's

File: pom.xml [hunk 5]
```diff
@@ -1052,6 +1054,13 @@
                 <version>${swagger.models.version}</version>
             </dependency>
 
+            <dependency>
+                <groupId>org.wso2.carbon.analytics</groupId>
+                <artifactId>org.wso2.carbon.business.rules.core</artifactId>
+                <version>${carbon.analytics.version}</version>
+            </dependency>
+
+
```
Comment (TanyaM) [inline pom.xml]: Unnecessary new lines

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/services/TemplateManagerService.java (hunk 2 above)
Comment (AnuGayan) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/services/TemplateManagerService.java]: Reimplemented by b181384b3c4bdd1b2ac145955cb54a71628805e7

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/services/TemplateManagerService.java (hunk 3 above)
Comment (AnuGayan) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/services/TemplateManagerService.java]: fixed by 3739e9380f6df941a58af8c9d43ae6c1d7eb7d6a

File: pom.xml (hunk 5 above)
Comment (AnuGayan) [reply pom.xml]: This was removed and readded by 71a1fc6a259ce78d8aba76b1f4c0a93a5f54ddda

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/bean/businessRulesFromTemplate/BusinessRuleFromTemplate.java (hunk 4 above)
Comment (AnuGayan) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/bean/businessRulesFromTemplate/BusinessRuleFromTemplate.java]: Fixed from b53d410a315b24891f480ea4b7e851caddb2e6cf

**Pull request merged**